### PR TITLE
Attempting to auto

### PR DESCRIPTION
**Pull request by calculusrobotics**

### CODE REVIEW

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java [hunk 1]
```diff
@@ -474,6 +503,14 @@ public double getRightDistance_meters() {
         return ((config.drive.invertRightCommand) ? -1 : 1) * rightMotors[0].getSelectedSensorPosition() * DRIVE_UTILS.WHEEL_CIRCUMFERENCE_INCHES / (config.drive.gearRatio * config.drive.ticksPerRevolution) * DriveConstants.METERS_PER_INCH;
     }
 
+    public double getLeftVelocity_mps() {
+        return ((config.drive.invertLeftCommand) ? -1 : 1) * 10 * leftMotors[0].getSelectedSensorVelocity() * DRIVE_UTILS.WHEEL_CIRCUMFERENCE_INCHES / (config.drive.gearRatio * config.drive.ticksPerRevolution) * DriveConstants.METERS_PER_INCH;
+    }
+
+    public double getRightVelocity_mps() {
+        return ((config.drive.invertRightCommand) ? -1 : 1) * 10 * rightMotors[0].getSelectedSensorPosition() * DRIVE_UTILS.WHEEL_CIRCUMFERENCE_INCHES / (config.drive.gearRatio * config.drive.ticksPerRevolution) * DriveConstants.METERS_PER_INCH;
```
Comment (cpostbitbuckets) [inline InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java]: This shouldn’t be sensor position.

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java [hunk 2]
```diff
@@ -99,24 +111,27 @@ public DriveSubsystem(Config config, NavigationSubsystem navigationSubsystem, OI
 
         DifferentialDriveKinematicsConstraint kinematicsConstraint = new DifferentialDriveKinematicsConstraint(
             DRIVE_UTILS.KINEMATICS,
-            config.drive.maxAllowedSpeed_ips * DriveConstants.METERS_PER_INCH
+            config.auto.cruiseSpeed_mps * DriveConstants.METERS_PER_INCH
```
Comment (cpostbitbuckets) [inline InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java]: This is already in meters.

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java [hunk 3]
```diff
@@ -99,24 +111,27 @@ public DriveSubsystem(Config config, NavigationSubsystem navigationSubsystem, OI
...
+            config.auto.maxAcceleration_mps
         );
         trajectoryConfig.addConstraint(voltageConstraint);
         trajectoryConfig.addConstraint(kinematicsConstraint);
 
         autoTrajectory = TrajectoryGenerator.generateTrajectory(
-            new Pose2d(new Translation2d(0, 0), Rotation2d.fromDegrees(0)),
-            List.of(new Translation2d(1, 1), new Translation2d(2, -1)),//FieldConstants.OUR_POWER_CELL_1, FieldConstants.OUR_POWER_CELL_2),
-            new Pose2d(new Translation2d(3, 0), Rotation2d.fromDegrees(0)),
+            new Pose2d(FieldConstants.FRONT_OF_POWER_PORT, Rotation2d.fromDegrees(90)),
```
Comment (cpostbitbuckets) [inline InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java]: If we start at -90 we don’t want to rotate to 90. That would put our intake the wrong direction.

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java (hunk 2 above)
Comment (calculusrobotics) [reply InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java]: fixed!

File: InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java (hunk 3 above)
Comment (calculusrobotics) [reply InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java]: The NavX has angles backward so I offset it by -90 then when reading the value negate it again, so WPI code should already think its at 90

File: InfiniteRecharge/src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -71,19 +71,19 @@ public void robotInit() {
...
 
+        if (config.enableShooterSubsystem) {
+            shooterSubsystem = new ShooterSubsystem(config, visionSubsystem);
+            subsystems.add(shooterSubsystem);
+        }
+
         if (config.enableDriveSubsystem) {
             navigationSubsystem = new NavigationSubsystem(config, visionSubsystem);
-            driveSubsystem = new DriveSubsystem(config, navigationSubsystem, oi);
+            driveSubsystem = new DriveSubsystem(config, navigationSubsystem, shooterSubsystem, oi);
```
Comment (cpostbitbuckets) [inline InfiniteRecharge/src/main/java/frc/robot/Robot.java]: I don't think it's right to make the shooterSubsystem a dependency on the driveSubsystem. The driveSubsystem should use the vision subsystem, not the shooter. All the driveSubsystem is getting from the shooter is the tx from the vision.

**Comment by cpostbitbuckets** [comment]
There doesn't seem to be any code that switches into ALIGN drive mode. It should switch when the user is holding down the aimBot() button.

File: InfiniteRecharge/src/main/java/frc/robot/Robot.java (hunk 4 above)
Comment (calculusrobotics) [reply InfiniteRecharge/src/main/java/frc/robot/Robot.java]: my reasoning was that we want the filtered value that the shooter subsystem has, but I just realized I'd need some (option to) filter with yaw in the same way that the shooter filters using the shooter position. will give it some more thought and see what I come up with

**Comment by calculusrobotics** [comment]
oops, I put in state change code but hadn't pushed it! try that now

**Pull request merged**